flask_api.py

```diff
@@ -117,14 +128,46 @@ def save_a_recording():
     return filename
 
 
-@app.route('/new_data/courses', methods=['POST'])
+@app.route("/new_data/phrase", methods=["POST"])
+def save_query_phrase():
+    validator = PhrasesValidator()
+    data = request.get_json()
+    try:
+        issues = validator.validate(data)
+    except:
+        return (
+            "Please format the query data: {question: {text: string, variables: list}, answer: {text: string, variables: list}}",
+            BAD_REQUEST,
+        )
+    if issues:
+        try:
+            data = validator.fix(data, issues)
+        except PhrasesValidatorError as err:
+            print("error", err)
+            return str(err), BAD_REQUEST
+
+    db = NimbusMySQLAlchemy(config_file=CONFIG_FILE_PATH)
+    try:
+        phrase_saved = db.insert_entity(QuestionAnswerPair, data)
+    except (BadDictionaryKeyError, BadDictionaryValueError, NimbusDatabaseError) as e:
+        return str(e), BAD_REQUEST
```

Comment:
Is it necessarily a bad request if there was an error saving the phrase? What if the database connection is broken (NimbusDatabaseError?), then we should probably 500.

Reply:
Good point, I'll add that status code

Reply:
I'll keep the first two as bad request, since that has to do with the content of the request